e2e/cypress/integration/enterprise/permissions/team_permissions_spec.js

```diff
@@ -0,0 +1,223 @@
+
+const deleteExistingTeamOverrideSchemes = () => {
+    cy.apiGetSchemes('team').then(({schemes}) => {
+        schemes.forEach((scheme) => {
+            cy.apiDeleteScheme(scheme.id);
+        });
+    });
+};
+
+const createTeamOverrideSchemeWithPermission = (name, team, permissionId, permissionValue) => {
+    cy.apiAdminLogin();
+
+    // # Go to `User Management / Permissions` section
+    cy.visit('/admin_console/user_management/permissions');
+
+    // # Click `New Team Override Scheme`
+    cy.findByTestId('team-override-schemes-link').should('be.visible').click().wait(TIMEOUTS.HALF_SEC);
+
+    // # Type Name and Description
+    cy.get('#scheme-name').should('be.visible').type(name);
+    cy.get('#scheme-description').type('Description');
+
+    // # Click `Add Teams`
+    cy.findByTestId('add-teams').should('be.visible').click().wait(TIMEOUTS.HALF_SEC);
+
+    // # Find and select testTeam
+    cy.get('#selectItems input').type(team.display_name).wait(TIMEOUTS.HALF_SEC);
+    cy.get('#multiSelectList div.more-modal__row.clickable').eq(0).click().wait(TIMEOUTS.HALF_SEC);
+
+    // # Save scheme
+    cy.get('#saveItems').should('be.visible').click().wait(TIMEOUTS.HALF_SEC);
+
+    // # Modify the permissions scheme
+    cy.findByTestId(permissionId).then((el) => {
+        if ((!el.hasClass('checked') && permissionValue) || (el.hasClass('checked') && !permissionValue)) {
+            el.click();
+        }
+    });
+
+    // # Save scheme
+    cy.get('#saveSetting').click().wait(TIMEOUTS.TWO_SEC);
+    cy.apiLogout();
+};
+
+describe('Team Permissions', () => {
+    let testTeam;
+    let testUser;
+    let testPrivateCh;
+    let otherUser;
+    const schemeName = 'schemetest';
+    before(() => {
+        cy.apiRequireLicense();
+        cy.apiInitSetup().then(({team, user}) => {
+            testTeam = team;
+            testUser = user;
+            cy.apiCreateChannel(testTeam.id, 'private-permissions-test', 'Private Permissions Test', 'P', '').then(({channel}) => {
+                cy.apiAddUserToChannel(channel.id, testUser.id);
+                testPrivateCh = channel;
+            });
+            cy.apiCreateUser().then(({user: newUser}) => {
+                otherUser = newUser;
+                cy.apiAddUserToTeam(testTeam.id, otherUser.id);
+            });
+        });
+    });
+
+    beforeEach(() => {
+        cy.apiLogout();
+        cy.apiAdminLogin();
+        cy.apiResetRoles();
+        deleteExistingTeamOverrideSchemes();
+    });
+
+    it('MM-T2871 Member cannot add members to the team', () => {
+        createTeamOverrideSchemeWithPermission(schemeName, testTeam, 'all_users-teams_team_scope-send_invites-checkbox', false);
+        cy.apiLogin(testUser);
+
+        // # Go to main channel
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Open hamburger menu
+        cy.findByLabelText('main menu').should('be.visible').click().wait(TIMEOUTS.HALF_SEC);
```

Comment:
fyi, there's now `cy.uiOpenMainMenu()` that does the same action. This can remain as is but will eventually change using such convenient command.